api/src/main/java/com/netflix/iceberg/TableScan.java

```diff
@@ -66,6 +66,17 @@
    */
   TableScan project(Schema schema);
 
+  /**
+   * Create a new {@link TableScan} from this that, if data columns where selected
+   * via {@link #select(java.util.Collection)}, controls whether the match to the schema will be done
+   * with case sensitivity.
+   *
+   * MUST be called before {@link #select(java.util.Collection)}!
```

Comment:
This seems like a problem to me. Can we do everything lazily so that it doesn't matter what order refinement calls are made in?

Reply:
Right, that is a reasonable expectation. Will fix.